### ISSUE TITLE
Bring back `skimage.test`

### ISSUE THREAD

**Issue by stefanv** [scikit-image/scikit-image#3569]
This turns out to be very useful for users who simply do a pip install and want to test.

/cc @hmaarrfk

**Comment by hmaarrfk** [comment]
I think this is actually fixed by putting the conftest.py file in the installed skin age directory. This has worked on my local machine. The respective PRs seem to have typos in the Travis files.

**Comment by hmaarrfk** [comment]
xref: https://github.com/scikit-image/scikit-image/pull/3311
This PR moves the `conftest.py` to the skimage directory so it gets installed on the user's machine.

**Comment by rfezzani** [comment]
I assume this is fixed by #3311

**Comment by stefanv** [comment]
@hmaarrfk Maybe you can clarify the comment above; the issue asks for `skimage.test()` to be re-introduced.

**Comment by stefanv** [comment]
Thanks, Greg!

**Comment by hmaarrfk** [comment]
I tihnk combined with @grlee77 's changes this should work.

The problem is that sometimes tools assume that certain files outside the `skimage` subfolder exist.
However, we currently test with `(cd .. && pytest --pyargs skimage)` so most of those concerns may have already been addressed.

Most of the questions we see refer to serious installation problems that transcend what can be debugged from a python console.
Users get stuck at the `import skimage` step. They simply wouldn't be able to run
```python
import skimage
simage.test()
```

Users interested in installing skimage on esoteric machines (High Performance Computing clusters, mobile) will generally be comfortable with the terminal and can use the `pytest --pyargs skimage` command. It is the same one we use to test scikit-image on the build:
https://github.com/scikit-image/scikit-image/blob/main/tools/github/script.sh#L17

This ensures we don't duplicate functionality built into pytest.

So to summarize:
1. If you think that asking users to run
```
import skimage
skimage.test()
```

is useful, then the code that @grlee77 is suggesting is useful.

2. If you think that it is just extra maintenance burden for us, then I think we should stick with `pytest --pyargs skimage`.


I prefer option `#2`. But I might be missing something critical.

**Comment by stefanv** [comment]
Oh, yes, good idea!
